analyzer.go

```diff
@@ -49,12 +49,9 @@ func (a *Analyzer) Analyze() (platform.AnalyzedMetadata, error) {
 			appMeta = platform.LayersMetadata{}
 		}
 
-		if a.Platform.API().AtLeast("0.8") {
-			if appMeta.BOM != nil && appMeta.BOM.SHA != "" {
-				a.Logger.Infof("Restoring data for sbom from previous image")
-				if err := a.SBOMRestorer.RestoreFromPrevious(a.PreviousImage, appMeta.BOM.SHA); err != nil {
-					return platform.AnalyzedMetadata{}, errors.Wrap(err, "retrieving launch sBOM layer")
-				}
+		if bomSHA(appMeta) != "" {
```

Comment:
Forgive me for [repeating myself](https://github.com/buildpacks/lifecycle/pull/773#discussion_r761418935) but might as well move all the `if`, if there's a brand new object.

Reply:
I started to implement your suggestion with gusto, but now I'm more on the fence about it. Removing the `if` would mean giving `SBOMRestorer` knowledge of `platform.LayersMetadata` (and probably also `platform.CacheMetadata`) which it doesn't really need. It would be nice to be able to change the schema of these labels and leave the restorer's behavior unaffected.

Reply:
That’s totally sound rationale. I can’t argue with that 🙂

Reply:
I think I confused myself... we could move the empty check inside the service (we'd still pass it `bomSHA(appMeta)`). 

> To answer your original question: I like this “Null Object” pattern a lot 👍🏾

I'm so happy!

Reply:
> we could move the empty check inside the service

Please see 9ad2d509